retrofit2/src/main/java/com/linecorp/armeria/client/http/retrofit2/ArmeriaCallFactory.java

```diff
@@ -36,26 +47,49 @@
 import okio.Buffer;
 
 /**
- * A {@link Call.Factory} that creates a {@link Call} instance for {@link HttpClient}.
+ * A {@link Factory} that creates a {@link Call} instance for {@link HttpClient}.
  */
-public class ArmeriaCallFactory implements Call.Factory {
-
-    private final HttpClient httpClient;
-
-    /**
-     * Creates a {@link Call.Factory} using the specified {@link HttpClient} instance.
-     *
-     * @param httpClient The {@link HttpClient} instance to be used.
-     */
-    public ArmeriaCallFactory(HttpClient httpClient) {
-        this.httpClient = httpClient;
+final class ArmeriaCallFactory implements Factory {
+
+    static final String GROUP_PREFIX = "group_";
+    private static final Pattern GROUP_PREFIX_MATCHER = Pattern.compile(GROUP_PREFIX);
+    private final Map<String, HttpClient> httpClients = new ConcurrentHashMap<>();
+    private final HttpClient baseHttpClient;
+    private final ClientFactory clientFactory;
+    private final BiFunction<String, ? super ClientOptionsBuilder, ClientOptionsBuilder> configurator;
+    private final String baseAuthority;
+
+    ArmeriaCallFactory(HttpClient baseHttpClient,
+                       ClientFactory clientFactory,
+                       BiFunction<String, ? super ClientOptionsBuilder, ClientOptionsBuilder> configurator,
+                       String groupPrefix) {
+        this.baseHttpClient = baseHttpClient;
+        this.clientFactory = clientFactory;
+        this.configurator = configurator;
+        baseAuthority = baseHttpClient.uri().getAuthority();
+        httpClients.put(baseAuthority, baseHttpClient);
     }
 
     @Override
     public Call newCall(Request request) {
         return new ArmeriaCall(this, request);
     }
 
+    private static boolean isGroup(String authority) {
+        return authority.startsWith(GROUP_PREFIX);
+    }
+
+    private HttpClient getHttpClient(String authority, String sessionProtocol) {
```

Comment:
```java
if (baseAuthority.equals(authority)) {
  return baseHttpClient;
}
```
?

Small optimization for the 99.9% case where there's only ever one authority. Otherwise, we don't need the fields for baseAuthority and baseHttpClient.